src/notes/notes.service.ts

```diff
@@ -333,6 +346,104 @@ export class NotesService {
     return note.owner;
   }
 
+  /**
+   * @async
+   * Add the specified alias to the note.
+   * @param {Note} note - the note to add the alias to
+   * @param {string} alias - the alias to add to the note
+   * @throws {AlreadyInDBError} the alias is already in use.
+   */
+  async addAlias(note: Note, alias: string): Promise<Note> {
```

Comment:
This should check if the new alias is already used as a noteId somewhere, to prevent https://github.com/hedgedoc/hedgedoc/issues/101